conda/install.py

```diff
@@ -1046,7 +1046,8 @@ def messages(prefix):
     path = join(prefix, '.messages.txt')
     try:
         with open(path) as fi:
-            sys.stdout.write(fi.read())
+            fh = sys.stderr if context.json else sys.stdout
+            fh.write(fi.read())
```

Comment:
If this weren't needed as a quick patch, I'd actually transition it to using loggers properly.  `TODO` though.